src/lib/Imp.hs

```diff
@@ -737,6 +738,17 @@ makeAllocDest allocTy ty = do
     return (ptr @> toScalarAtom ptr')
   impSubst env dest
 
+-- TODO: deallocation!
+makeAllocDestForPtr :: Type -> ImpM (Dest, Atom)
+makeAllocDestForPtr ty = do
+  (ptrSizes, dest) <- fromEmbed $ makeDest (LLVM, CPU, Unmanaged) ty
+  case ptrSizes of
+    [(Bind (ptr:>PtrTy ptrTy), size)] -> do
+      ptr' <- emitAlloc ptrTy $ fromScalarAtom size
+      dest' <- impSubst (ptr @> toScalarAtom ptr') dest
+      return (dest', toScalarAtom ptr')
+    _ -> error $ "expected a single pointer"
```

Comment:
I'm really not a fan of committing functions that leak memory, because at some point we will want to fix all those leaks, and it will be a pain to track them all down again 😕 We should really have `bracket` style functions for allocating pointers for a region of code.